Qualitative/BioGrid-based predictions of regulatory relationships

NElnour opened the pull request:
These tools are for BioGrid-based predictions of regulatory relationships using an interaction interpretation map developed on both GGI and PPI. I still need to develop another interaction interpretation map based on GGI only.

Commit:
added scripts for integrating BioGrid GGI and STRING PPI data of system into directed networks

Commit:
annotated scripts with authorships

Commit:
edited readme with specifications

Commit:
added option for user to display stringent vs relaxed interaction network

Commit:
visualize relaxed networks

Commit:
embedded network view

Commit:
fixed readme

Commit:
relaxed network view

Commit:
added network view

Commit:
updated README for ease of integration

Commit:
Update README.md

Commit:
fixed biogridr credentials

Commit:
wrote getKey to set up user credentials

Commit:
Merge branch 'master' of https://github.com/NElnour/BCB420.2019.ESA

Commit:
added GGI interpretation map

Commit:
attempt to resolve Travis build failure

Commit:
replaced with original files

Commit:
resolved warnings

Commit:
added import statements to Description

Commit:
added roxygen2-formatted documentation

Commit:
resolved roxygen2 dependencies

Commit:
removed old .Rd files

Commit:
removed examples from .Rd files

Commit:
renamed tool

Commit:
removed old .rd files

Commit:
Resolved roxygen2 errors and warnings

Commit:
updated documentation with import statements

Commit:
added import support for remote package biogridr

Commit:
resolved import clashes

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
network hypothesis

Commit:
Update README.md

Commit:
refactored code to remove dependency on external datasets; generated updated .Rd files

Commit:
fixed ratification problem

Commit:
reverted biogridr changes

Commit:
fixed example building issues

Commit:
updated readme

Commit:
multisystem example added

Commit:
updated readme annotations

Commit:
fixed readme example

Commit:
reolved merge conflicts

Merge branch 'master' of git://github.com/hyginn/BCB420.2019.ESA

# Conflicts:
#	NAMESPACE
#	R/SyDButils.R
#	R/fetchComponents.R
#	R/fetchData.R
#	README.md

Commit:
removed merge markers

sapirl commented on DESCRIPTION:
```diff
@@ -19,11 +23,30 @@ Encoding: UTF-8
+Remotes: NElnour/biogridr
 RoxygenNote: 6.1.1
 VignetteBuilder: knitr
+Collate: 
```
Please check where the following changes come from. I am not sure what "Collate" denotes for, but all of the files besides "GGI_PPI_Integrator.R" should not be shown as altered.

NElnour replied:
Collation describes the order with which the different files need to be built for my tool to work. I don't think it indicates that these files are necessarily altered.

hyginn replied:
Requiring collation is often a sign of a brittle code architecture. Can you remove the need to load in a particular sequence, or elaborate why this is necessary? Moreover, a `Collate:` statement in the `DESCRIPTION` affects all co-authors...

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' myKey <- getKey("Nada Elnour", "nada.elnour@@mail.utoronto.ca", "SILGRESA")
+#' mySys <- getSysInteractions("SLIGR", key = myKey, criterion = "stringent")
+#'
+#' @export
```
If I understood correctly, the "@export" should only appear once per file. I.e., If you want to export multiple functions, then each exported function should be in a separate file; If you want to export only one function, the "@export" should appear only once, for the exported function.

hyginn replied:
Generally that is correct ... but sometimes exceptions are useful, e.g. if they are semantically very similar functions. The issue is: how easy is it to find the code for a function in the ./R directory. E.g. all my `SyDB...()` functions are in one file.

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' @import biogridr
+#' @import utils
+#' @importFrom stats complete.cases
+#' @export
```
Multiple "@export" in one file

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' @examples
+#' EMAP <- makeEMAP()
+#'
+#' @export
```
Multiple "@export" in one file

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' @examples
+#' GMAP <- makeGMAP()
+#'
+#' @export
```
Multiple "@export" in one file

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' hypothesize(mySys)
+#' hypothesize(mySys2, mySys)
+#'
+#' @export
```
Multiple "@export" in one file

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' @param gmap The dataframe obtained from \code{makeGMAP}.
+#' @param emap The dataframe obtained from \code{makeEMAP}.
+#'
+#' @export
```
Multiple "@export" in one file. An helper function should not be exported

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+#' @examples
+#' myKey <- getKey("Nada Elnour", "nada.elnour@@mail.utoronto.ca", "SILGRESA")
+#'
+#' @export
```
Multiple "@export" in one file

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+  library(biogridr)
+}
+
+# Load all required packages.
```
If I understood correctly, it is not necessary to load the packages because they will be imported due to the fact that you listed them under "imports"

sapirl commented on R/GGI_PPI_Integrator.R:
```diff
@@ -0,0 +1,415 @@
+      stop("Provide valid key for biogrid ratification")
+    }
+
+    myURL <-
```
The string edges data could be fetched by the "fetchdata" function, and the data itself is downloaded automatically whenever someone installs the package (so there is no need to download it again)

sapirl commented on R/fetchComponents.R:
```diff
@@ -15,29 +15,108 @@ fetchComponents <- function(sys) {
   # returns a fixed set of symbols.
   # Function stub for development purposes only.
 
-  message("Note: fetchComponents(...) is deprecated.",
```
There are some changes within Dr. Steipe's functions

NElnour replied:
Hey, I can't find an inconsistency between the two files save for formatting.

hyginn replied:
... and that `fetchComponents()` produces the "deprecated" warning. But if possible, switch to `SydDBgetSysSymbols()`.

hyginn replied:
Ideally, just pull the updated version, then the file should not even appear here.

Commit:
refactored exported functions into separate files

hyginn commented on R/getKey.R:
```diff
@@ -0,0 +1,29 @@
+#' @return The unique key for user and project for data retrieval from BioGrid
+#' @import biogridr
+#' @examples
+#' myKey <- getKey("Nada Elnour", "nada.elnour@@mail.utoronto.ca", "SILGRESA")
```
Don't ever post valid credentials anywhere on GitHub. Even if these are not valid, it's better to use dummy names and variables. See [section 2.6.7 in the rptPlus README](https://github.com/hyginn/rptPlus#267-a-secure-approach-to-credentials).

hyginn commented on DESCRIPTION:
```diff
@@ -19,11 +23,34 @@ Encoding: UTF-8
     BiocStyle,
     knitr,
     rmarkdown
+Remotes: NElnour/biogridr
```
Oh no ... that's an external dependency. We really should not do that. Why do you you need this package?

NElnour replied:
Hello, 

biogridr allowed me to get genetic interaction data from BioGrid before fetchData was available. I'll remove it.

hyginn commented on R/getGeneticInteractome.R:
```diff
@@ -0,0 +1,62 @@
+#' @param criterion A string, either "stringent" or "relaxed", specifying whether only GGI between physical interactors should be selected.
+#' @return The dataframe of system components and either their GGI between physical interactors should be selected (iff \code{criterion} == "stringent")
+#' or all GGI of physical interactors (iff \code{relaxed} == "stringent")
+#' @import xlsx
```
Do you really need all of these `@import`'s for the function? The packages are available, since they are mentioned in `DESCRIPTION`: you should be able to use `<package>::<function>()`. I look at the code, and see that I would need to do a lot of searching until I find wher`bg()` comes from, just as one example.

hyginn commented on R/getGeneticInteractome.R:
```diff
@@ -0,0 +1,62 @@
+    )))
+
+  myGenes <- gsub(", ", "|", myGenes)
+  humInt <-
```
I'm not a fan of `%>%` at all since it is much harder to debug functions when you can't inspect intermediate values.  But this is of course technically valid code. In situations such as these I usually write something like:
```
x <- f1()
x <- f2(x)
x <- f3(x)
```

NElnour replied:
Fixed.

hyginn commented on R/getGeneticInteractome.R:
```diff
@@ -0,0 +1,62 @@
+#' @import utils
+#' @importFrom stats complete.cases
+#' @export
+getGeneticInteractome <- function(mySys, criterion, key) {
```
There are no comments in this code  :-(

NElnour replied:
Fixed.

hyginn commented on R/getKey.R:
```diff
@@ -0,0 +1,29 @@
+# getKey.R
```
This is not a "script" but a function. Scripts should go into `./inst/scripts`.

NElnour replied:
I removed the function. Thanks

hyginn commented on R/hypothesize.R:
```diff
@@ -0,0 +1,129 @@
+# hypothesize.R
+
+#' Generates an annotated hypothesis graph whose nodes are components of \code{mySys} and edges are GGI interpretations in either \code{EMAP} or \code{GMAP}
```
Keep line lengths under 80 character. `BiocCheck()` would complain.

NElnour replied:
Fixed.

hyginn commented on R/hypothesize.R:
```diff
@@ -0,0 +1,129 @@
+# hypothesize.R
```
I don't understand what's happening here, without comments. There's a lot of to-and-fro between strings and factors and I'm not sure that's necessary...

NElnour replied:
Hello, I simplified the function

hyginn commented on R/predictDirectedNetworks.R:
```diff
@@ -0,0 +1,76 @@
+# Dependencies: tibble 2.0.1; biomaRt 2.38.0; xlsx 0.6.1; readxl 1.3.1; dplyr 0.8.0.1; ggplot2 3.1.0; biogridr 0.0.0.9000; visNetwork 2.0.5.
+
+# ====  GLOBAL PACKAGES ========================================================
+#' @import devtools
```
No - don't use `devtools`. The functions are part of the package.

Commit:
changed formatting per reviewer's suggestions

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	README.md

Commit:
removed unecessary imports and refactored getSysInteractions to remove biogridr dependency

Commit:
added comments to functions

Commit:
removed collation

Commit:
updated documentation and simplified hypothesize.R

Commit:
I think someone installed my fork and made changes. I resolved the merge conflict.

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE
